### PR TITLE
Add: add black lint for the repo

### PR DESCRIPTION
**Pull request by TienanLi**
ATT.

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (shineyruan) [inline .pre-commit-config.yaml:1]: Add black lint to pre-commit?

**Review by shineyruan — DISMISSED** [review]

**Review by shineyruan — APPROVED** [review]

**Comment by shineyruan** [comment]
can we also add require pre-commit.ci to pass before PR merge?

**Comment by TienanLi** [comment]
> can we also add require pre-commit.ci to pass before PR merge?

Somehow I can't see the option of pre-commit.ci in the droplist yet... Is it because we have to wait a few days after the first time usage?...
Screenshot from 2024-10-20 17-58-16

**Comment by shineyruan** [comment]
> > can we also add require pre-commit.ci to pass before PR merge?
> 
> Somehow I can't see the option of pre-commit.ci in the droplist yet... Is it because we have to wait a few days after the first time usage?... Screenshot from 2024-10-20 17-58-16

You might have to install pre-commit.ci plugin in your Github account first. Did you sign with your Github account on [pre-commit.ci](https://pre-commit.ci/) ?

**Comment by TienanLi** [comment]
> > > can we also add require pre-commit.ci to pass before PR merge?
> > 
> > 
> > Somehow I can't see the option of pre-commit.ci in the droplist yet... Is it because we have to wait a few days after the first time usage?... Screenshot from 2024-10-20 17-58-16
> 
> You might have to install pre-commit.ci plugin in your Github account first. Did you sign with your Github account on [pre-commit.ci](https://pre-commit.ci/) ?

yes! that works now

**Pull request merged**